bin/pytorch_inference/CCommandParser.cc

```diff
@@ -81,60 +85,104 @@ bool CCommandParser::ioLoop(const TRequestHandlerFunc& requestHandler,
             return false;
         }
 
-        if (validateJson(doc, errorHandler) == false) {
+        EMessageType messageType = validateJson(doc, errorHandler);
+        LOG_TRACE(<< "Inference command: " << doc);
+        if (messageType == EMessageType::E_MalformedMessage) {
             continue;
         }
 
-        LOG_TRACE(<< "Inference command: " << doc);
-        jsonToRequest(doc);
-        if (requestHandler(m_Request) == false) {
-            LOG_ERROR(<< "Request handler forced exit");
-            return false;
+        if (messageType == EMessageType::E_InferenceRequest) {
+            jsonToInferenceRequest(doc);
+            if (requestHandler(m_Request) == false) {
+                LOG_ERROR(<< "Request handler forced exit");
+                return false;
+            }
+        } else if (messageType == EMessageType::E_ControlMessage) {
+            jsonToControlMessage(doc);
+            controlHandler(m_ControlMessage);
         }
     }
 
     return true;
 }
 
-bool CCommandParser::validateJson(const rapidjson::Document& doc,
-                                  const TErrorHandlerFunc& errorHandler) const {
+CCommandParser::EMessageType
+CCommandParser::validateJson(const rapidjson::Document& doc,
```

Comment:
I suspect that all these methods can be static. Probably slightly cleaner if they don't use any state.